SummerProject/Assets/Scripts/Enemy/MeleeEnemy.cs

```diff
@@ -64,37 +76,73 @@ public class MeleeEnemy : MonoBehaviour, IEnemy {
     }
 
     /**
-     * Simply move towards the player
+     * Simply move towards the player using navmesh
      */
     public void Move() {
         ThisAgent.SetDestination(target.position);
+        SetWalkingAnimation();
     }
 
     /**
      * Play the death animation, then delete the model after it finishes
      */
     public void Death() {
         Animation death = GetComponent<Animation>(); // Need to get actual animations lol
-        death.Play();
+        animator.SetTrigger("Death");
         Destroy(gameObject, death.clip.length);
     }
 
     /**
-     * Checks if enemy is in range of player to perform an attack
+     * Checks to see if the melee cool down has refreshed or not
      */
-    private bool InRange() {
-        if (Vector3.Distance(target.position, transform.position) <= meleeDistance) {
-            return true;
-        }
-
-        return false;
-    }
-
     private bool canAttack() {
         // Add animation duration later
         return (Time.time - meleeStartTime) > meleeCD;
     }
 
+    /**
+     * Returns Distance between enemy and player
+     */
+    private float CheckPlayerDistance() {
+        return Vector3.Distance(target.position, transform.position);
+    }
+
+    /**
+     * Sets the corresponding walking animation based on angle to player
+     */
+    private void SetWalkingAnimation() {
+        Vector3 targetDir = target.position - transform.position;
+        float angle = Vector3.SignedAngle(targetDir, transform.forward, Vector3.up);
+
+        if (angle <= 43.75 && angle >= -43.75) {
```

Comment:
store the number for angles in variable (better for style, clarity, and easier to change)